src/attestation/sev.rs

```diff
@@ -84,46 +97,97 @@ mod tests {
 
     use super::*;
 
+    fn test_message_representation(bytes: &[u8], expected: Message) {
+        let actual: Message = serde_flavor::from_reader(bytes).unwrap();
+        assert_eq!(actual, expected);
+
+        let mut out = vec![];
+        serde_flavor::to_writer(&mut out, &actual).unwrap();
+        let actual: Message = serde_flavor::from_reader(&out[..]).unwrap();
+        assert_eq!(actual, expected);
+    }
+
     #[test]
-    fn test_message_representation() {
-        let test_cases = &[
-            // {"mimetype": "certificate-chain-naples", "payload": [1, 2, 3, 4]}
-            Message {
-                mimetype: MimeType::CertificateChainNaples,
-                payload: vec![1, 2, 3, 4],
-            },
-            // {"mimetype": "certificate-chain-rome", "payload": [1, 2, 3, 4]}
-            Message {
-                mimetype: MimeType::CertificateChainRome,
-                payload: vec![1, 2, 3, 4],
-            },
-            // {"mimetype": "launch-start", "payload": [1, 2, 3, 4]}
-            Message {
-                mimetype: MimeType::LaunchStart,
-                payload: vec![1, 2, 3, 4],
-            },
-            // {"mimetype": "measurement", "payload": [1, 2, 3, 4]}
-            Message {
-                mimetype: MimeType::Measurement,
-                payload: vec![1, 2, 3, 4],
-            },
-            // {"mimetype": "secret", "payload": [1, 2, 3, 4]}
-            Message {
-                mimetype: MimeType::Secret,
-                payload: vec![1, 2, 3, 4],
-            },
-            // {"mimetype": "finish", "payload": [1, 2, 3, 4]}
-            Message {
-                mimetype: MimeType::Finish,
-                payload: vec![1, 2, 3, 4],
-            },
-        ];
-
-        for test_case in test_cases {
-            let mut encoded = vec![];
-            serde_flavor::to_writer(&mut encoded, test_case).unwrap();
-            let actual = serde_flavor::from_reader(&encoded[..]).unwrap();
-            assert_eq!(*test_case, actual);
-        }
+    fn test_message_certificate_chain_naples() {
+        let input = b"\xA1\x78\x18\x63\x65\x72\x74\x69\x66\x69\x63\x61\x74\x65\
+                    \x2D\x63\x68\x61\x69\x6E\x2D\x6E\x61\x70\x6C\x65\x73\xA6\
+                    \x63\x61\x72\x6B\x84\x01\x02\x03\x04\x63\x61\x73\x6B\x84\
+                    \x05\x06\x07\x08\x63\x70\x65\x6B\x84\x09\x0A\x0B\x0C\x63\
+                    \x63\x65\x6B\x84\x0D\x0E\x0F\x10\x63\x70\x64\x68\x84\x11\
+                    \x12\x13\x14\x63\x6F\x63\x61\x84\x15\x16\x17\x18\x18";
+        let expected = Message::CertificateChainNaples(Chain {
+            ark: vec![1, 2, 3, 4],
+            ask: vec![5, 6, 7, 8],
+            pek: vec![9, 10, 11, 12],
+            cek: vec![13, 14, 15, 16],
+            pdh: vec![17, 18, 19, 20],
+            oca: vec![21, 22, 23, 24],
```

Comment:
How is it that the `Message` appears to have dummy values in its fields, but the `input` byte slice is something specific, and they are equivalent?

Reply:
I hand-crafted the message types on [cbor.me](http://cbor.me/) and had it generate the byte representation with the dummy values you see here, then I pasted those in as input strings. I thought it would be helpful to have an encoded "input" that was created _not_ using this code to ensure that everything works from end-to-end, including something that wasn't encoded by us.